File: Data Structures & Algorithms/is-anagram/submission-0.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.size() != t.size()){
            return false;
        }

        map<char, int> freq; 
        for(auto ch: s){
            freq[ch]++; 
        }

        for(auto ch: t){
            if(!freq.contains(ch)){
                return false; 
            }
            freq[ch]--;
            if(freq[ch] < 0){
                return false; 
            }
        }
        return true;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freq = collections.defaultdict(int) 

        for num in nums: 
            freq[num] += 1 

        sorted_freq = sorted(freq.items(), key=lambda item: item[1], reverse=True) 
        res = []
        for i in range(k): 
            res.append(sorted_freq[i][0]) 
        return res 